twilight-gateway/src/config.rs

```diff
@@ -11,10 +11,42 @@ use twilight_model::gateway::{
     Intents,
 };
 
-/// Identifier of a [shard], including its ID and the total number of shards in
-/// use by the bot.
+/// [`Shard`] identifier to calculate if it receivies a given event.
 ///
-/// [shard]: crate::Shard
+/// A shard ID consist of two fields: `number` and `total`. These values do not
+/// need to be unique, and are used by Discord for calculating which events to
+/// send to which shard. Shards should in general share the same `total` value
+/// and have an unique `number` value, but users may deviate from this when
+/// resharding/migrating to a new set of shards.
+///
+/// # Advanced use
+///
+/// Incoming events are split by their originating guild and are received by the
+/// shard with the id calculated from the following formula:
+///
+/// > `number = (guild_id >> 22) % total`.
```

Comment:
we could cite the discord docs here: https://discord.com/developers/docs/topics/gateway#sharding

Reply:
I linked to the docs below everything else as I thought it would otherwise break the reading flow.